WallsAndHoles/tilemaptoolmanager.h

```diff
@@ -12,8 +12,6 @@ class TileMapToolManager : public ToolManager
 public:
     TileMapToolManager(QObject *parent = nullptr);
 
-    QAction *registerTool(QSharedPointer<AbstractTileMapTool> tool, QString name) { return ToolManager::registerTool(tool, name); }
```

Comment:
Why remove this? It enforces the tool to be an AbstractTileMapTool as apposed to just a generic tool